vulcanai2/datasets/tabulardataset.py

```diff
@@ -319,46 +372,74 @@ def identify_unique(self, threshold):
 
     def identify_unbalanced_columns(self, threshold, non_numeric=True):
         """
-        This returns columns that are highly unbalanced, aka those that have a disproportionate amount of one value.
-        Return those columns that have at least threshold percentage of one value.
-        :param threshold: Proportion needed to define unbalanced, between 0 and 1
-        :param non_numeric: Whether non-numeric columns are also considered.
-        :return: The column list
+        This returns columns that are highly unbalanced.
+        Those that have a disproportionate amount of one value.
+
+        Parameters
+        ----------
+        threshold: Float
+            Proportion needed to define unbalanced, between 0 and 1
+        non_numeric: Boolean
+            Whether non-numeric columns are also considered.
+
+        Returns
+        -------
+        column_list: List
+            The list of column names
         """
         if non_numeric:
             columns = list(self.df.columns)
         else:
             columns = list(self.df.select_dtypes(include=np.number))
         column_list = []
         for col in columns:
```

Comment:
is there a way to do this without looping through the columns? perhaps pandas has some useful utilities?

Reply:
We might be able to use apply functions, but the readability might not be as nice. For example, `self.df.apply(pd.value_counts)`. However, I still believe looping to be the best option since we need to a couple operations on the columns.

Reply:
could you time it? if it is significantly slower than I would rather beef up the documentation to understand what is happening behind the scenes

Reply:
@rfratila So it looks like looping is faster than using apply method. However, using apply method is more concise. Here is the apply method:

```
col_max_val = self.df.apply(pd.value_counts).max()
ls_cols = col_max_val.where(col_max_val >= max_num_vals).dropna().index.values.tolist()
```

Reply:
interesting how looping is faster. in that case, no need to use the apply method.